### PR TITLE
Use scss module for styling Header Banner

### PR DESCRIPTION
**Pull request by megcrenshaw**
The original ask was to ensure that, on mobile, the text that is part of the header banner on Desktop stays inside the banner on mobile, rather than dropping below the banner.

There are 6 pages that use HeaderBanner:
image

I refactored the HeaderBanner component in a few ways:
1) I used Kelby's pattern on Pegasus as a guide (see https://github.com/code-dot-org/code-dot-org/pull/50929#issuecomment-1483014088)
2) I removed the `short` prop, instead opting for dynamic styling based on content. This is what the `short` prop was doing:
https://github.com/code-dot-org/code-dot-org/blob/de6ff638edb7ac5e06bb5dd2e0c5c34bbcb24c74/apps/src/templates/HeaderBanner.jsx#L12-L59
The ProjectHeader, the Courses banner (if not signed out), the Incubator, the StudentHomepage, and the TeacherHomepage all had the `short` prop set to `true`. See screenshots for the styling updates.
3) I also removed the responsiveSize prop, opting instead for media queries to determine styling in SCSS (also part of Kelby's styling mentioned in 1).

## Testing story (Screenshots)

Here are the screenshots for what's changed. I was *not* striving for parity, but rather ensuring responsiveness + readability of text for all header banners.

**Catalog**
image
image

Local:
image
image

--

**Projects**
image
image

Local:
image
image

--

**Courses (Signed Out)**
image
image

Local:
image
image

--

**Courses (Signed In)**
image
image

Local:
image
image

--

**Incubator**
image
image

Local:
image
image

--

**Teacher Homepage**
image
image

Local:
image
image

--

**Student Homepage**
image
image

Local:
image
image

RTL:

image
image
image
image
image

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by megcrenshaw** [comment]
@daynew and @breville looks like you have done a lot of work on the HeaderBanner, including on the pages mentioned above. Would love your eyes on this too!

**Review by davidsbailey — APPROVED** [review]
LGTM! Nice cleanup of tests!

File: apps/src/templates/header-banner.module.scss [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  padding: 2rem 1rem;
+  box-sizing: border-box;
+
+  /*
+    The banner background image leaves an open space for text on the left. Therefore, in Right-to-Left languages, the
+    banner needs to be flipped horizontally so the background image empty space is on the right.
+   */
+  html[dir="rtl"] & {
+    transform: scaleX(-1);
+  }
```
Comment (kelbyhawn) [inline apps/src/templates/header-banner.module.scss:17]: TIL! 😮

**Review by kelbyhawn — APPROVED** [review]
This looks great @megcrenshaw! Love the clean SCSS 👏 🎉

File: apps/src/templates/header-banner.module.scss (hunk 1 above)
Comment (megcrenshaw) [reply apps/src/templates/header-banner.module.scss:17]: YEAH, this was wild to learn about. And then having to "undo" the reversal on the text ... 🤯

Comment (kelbyhawn) [reply apps/src/templates/header-banner.module.scss:17]: I saved this in my Codebase Notes doc for future banners!

**Pull request merged**